### PR TITLE
Add stream crossover component

### PR DESCRIPTION
**Pull request by ethancheez**
| | |
|:---|:---|
|**_Originating Project/Creator_**| @ethancheez |
|**_Affected Component_**| StreamCrossover |
|**_Affected Architectures(s)_**| All |
|**_Related Issue(s)_**| N/A |
|**_Has Unit Tests (y/n)_**| No |
|**_Builds Without Errors (y/n)_**| Yes |
|**_Unit Tests Pass (y/n)_**| N/A |
|**_Documentation Included (y/n)_**| Yes |

---
## Change Description

Add stream crossover component as Drv::StreamCrossover

## Rationale

Allow connection from ByteStreamRecv to ByteStreamSend

## Testing/Review Recommendations

Implement unit tests.

### CODE REVIEW

**Comment by timcanham** [comment]
image

File: Drv/StreamCrossover/StreamCrossover.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  // ----------------------------------------------------------------------
+
+  void StreamCrossover ::
+    streamIn_handler(
+        const NATIVE_INT_TYPE portNum,
+        Fw::Buffer &recvBuffer,
+        const Drv::RecvStatus &recvStatus
+    )
+  {
+    this->streamOut_out(0, recvBuffer);
```
[the remark this answers is not in the document]
Comment (LeStarch) [reply Drv/StreamCrossover/StreamCrossover.cpp:42]: @ethancheez we should check these return values.  Either performing a re-try, or emitting a log or event message.

**Review by LeStarch — CHANGES_REQUESTED** [review]
We should do the following:
1. Fix this spelling error (add `ethanechee` to `expect.txt`)
2. Handle the send return value: emit an event.  
3. Could we add a UT to test this?

File: Drv/StreamCrossover/StreamCrossover.cpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  // ----------------------------------------------------------------------
+  // Handler implementations for user-defined typed input ports
+  // ----------------------------------------------------------------------
+
+  void StreamCrossover ::
+    streamIn_handler(
+        const NATIVE_INT_TYPE portNum,
+        Fw::Buffer &recvBuffer,
+        const Drv::RecvStatus &recvStatus
```
Comment (LeStarch) [inline Drv/StreamCrossover/StreamCrossover.cpp:39]: Receive status is not being checked.  We should probably check that and not send through if the status is bad.

File: Drv/StreamCrossover/docs/sdd.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+| Name | Description |
+|---|---|
+| streamOut | A ByteStreamSend port for outgoing data stored in `Fw::Buffer` |
+| streamIn  | A ByteStreamRecv port for incoming data stored in `Fw::Buffer` |
+
+## Requirements
+Add requirements in the chart below
+| Name | Description | Validation |
+|---|---|---|
+| STREAM-CROSSOVER-COMP-001 | The stream crossover component shall provide the capability to forward bytes | inspection |
```
Comment (LeStarch) [inline Drv/StreamCrossover/docs/sdd.md:23]: This requirement is now verified via unit-test.

File: Drv/StreamCrossover/test/ut/Tester.cpp [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  }
+
+  // ----------------------------------------------------------------------
+  // Tests
+  // ----------------------------------------------------------------------
+
+  void Tester ::
+    sendTestBuffer()
+  {
+    this->clearHistory();
```
Comment (LeStarch) [inline Drv/StreamCrossover/test/ut/Tester.cpp:37]: You could put the requirement call here to make it easier to trace.

**Review by LeStarch — APPROVED** [review]

**Review by thomas-bc — APPROVED** [review]

**Review by LeStarch — APPROVED** [review]

**Pull request merged**